common/docker/client.go

```diff
@@ -157,14 +157,35 @@ func (pending pendingStarts) finish(id string) {
 }
 
 func (c *Client) AllContainerIDs() ([]string, error) {
-	all, err := c.ListContainers(docker.ListContainersOptions{All: true})
+	return c.containerIDs()
+}
+
+// ExistingContainerIDs returns a list of container IDs which either are running or
+// might be started in the future.
+func (c *Client) ExistingContainerIDs() ([]string, error) {
+	return c.containerIDs("running", "paused", "exited", "created", "restarting")
```

Comment:
Why do we need this restriction?  Is it not possible for a "dead" container to restart?